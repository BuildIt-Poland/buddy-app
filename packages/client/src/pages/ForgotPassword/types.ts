import { RouteComponentProps } from 'react-router-dom';

export interface FormData {
  email: string;
}

export type ForgotPasswordProps = Pick<RouteComponentProps, 'history'>;
