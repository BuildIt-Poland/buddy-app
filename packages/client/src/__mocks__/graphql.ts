import { BUDDY_MENU_DETAILS, NEWBIE_MENU_DETAILS } from 'graphql/user-menu.graphql';
import { AVATAR_HEADER } from 'graphql/avatar-header.graphql';
import { NEWBIE_SELECT } from 'graphql/newbie-select.graphql';
import { BUDDY_SELECT } from 'graphql/buddy-select.graphql';
import { TASK_DETAILS } from 'graphql/task-details.graphql';
import { ADD_NEWBIE_TASK } from 'graphql/add-task.graphql';
import { ADD_NEWBIE } from 'graphql/add-user.graphql';
import { GraphQLError } from 'graphql';

import {
  MutationAddNewbieTaskArgs,
  AuthPayload,
  TaskStatus,
  Newbie,
  QueryNewbieArgs,
  QueryBuddyArgs,
  UserRole,
} from '@buddy-app/schema';
import {
  NEWBIE_CONTACT_DETAILS,
  BUDDY_CONTACT_DETAILS,
  TALENT_CONTACT_DETAILS,
} from 'graphql/contact-details.graphql';
import { TASK_LIST } from 'graphql/task-list.graphql';
import { getBasicUserDetailsMock } from './general';

export const loginSuccessMock = (data?: Partial<AuthPayload>) => ({
  data: {
    login: {
      role: 'BUDDY',
      token: 'dummy-token',
      userId: '1',
    },
  },
});

export const loginFailedMock = () => new GraphQLError('No such user found');

export const loginNoNetworkMock = () => ({
  networkError: true,
});

export const NewbieAvatarDetails = (variables?: QueryNewbieArgs) => ({
  request: {
    query: AVATAR_HEADER,
    variables,
  },
  result: {
    data: {
      newbie: {
        id: '1',
        name: 'Tom Hanks',
        position: 'front-end',
        photo: 'some-url',
      },
    },
  },
});

export const tasksResponse = (filter?: TaskStatus) => {
  const tasks = [
    {
      id: '1',
      title: 'Call you head manager',
      description: '',
      createdAt: '',
      implementationDate: '',
      status: TaskStatus.Uncompleted,
      newbie: {} as Newbie,
    },
    {
      id: '2',
      title: 'Write you main manager',
      description: '',
      createdAt: '',
      implementationDate: '',
      status: TaskStatus.Completed,
      newbie: {} as Newbie,
    },
  ];

  switch (filter) {
    case TaskStatus.Completed:
      return [tasks[0]];
    case TaskStatus.Uncompleted:
      return [tasks[1]];
    default:
      return tasks;
  }
};

export const buddyMenuDetails = (variables?: Partial<QueryBuddyArgs>) => [
  {
    request: {
      query: BUDDY_MENU_DETAILS,
      variables,
    },
    result: {
      data: {
        buddy: {
          id: '1234',
          name: 'Tom Hanks',
          email: 'tom@wipro.com',
          allowPushedNotifications: true,
          photo: 'buddy-photo-url',
          newbies: [
            {
              id: 'id',
              name: 'Sandra Bullock',
              photo: 'newbie-photo-url',
            },
          ],
        },
      },
    },
  },
];

export const newbieMenuDetails = (variables?: Partial<QueryNewbieArgs>) => [
  {
    request: {
      query: NEWBIE_MENU_DETAILS,
      variables,
    },
    result: {
      data: {
        newbie: {
          id: '1234',
          name: 'Sandra Bullock',
          email: 'doejoe@wipro.com',
          allowPushedNotifications: true,
          photo: 'newbie-photo-url',
          buddy: {
            id: 'id',
            name: 'Tom Hanks',
            photo: 'newbie-photo-url',
          },
        },
      },
    },
  },
];

export const taskListResponse = (
  variables: Partial<QueryNewbieArgs>,
  newbie: Partial<Newbie>
) => [
  {
    request: {
      query: TASK_LIST,
      variables,
    },
    result: {
      data: {
        newbie,
      },
    },
  },
];

export const newbieContactDetails = (variables?: Partial<QueryNewbieArgs>) => [
  {
    request: {
      query: NEWBIE_CONTACT_DETAILS,
      variables,
    },
    result: {
      data: {
        newbie: getBasicUserDetailsMock(UserRole.Newbie),
      },
    },
  },
];

export const buddyContactDetails = (variables?: Partial<QueryBuddyArgs>) => [
  {
    request: {
      query: BUDDY_CONTACT_DETAILS,
      variables,
    },
    result: {
      data: {
        buddy: getBasicUserDetailsMock(UserRole.Buddy),
      },
    },
  },
];

export const talentContactDetails = (variables?: Partial<QueryBuddyArgs>) => [
  {
    request: {
      query: TALENT_CONTACT_DETAILS,
      variables,
    },
    result: {
      data: {
        buddy: getBasicUserDetailsMock(UserRole.Talent),
      },
    },
  },
];

export const newbieSelectMock = [
  {
    request: {
      query: NEWBIE_SELECT,
      variables: {
        buddyId: 'ck17sl83c9gya0b17dcvttzm4',
      },
    },
    result: {
      data: {
        buddy: {
          id: 'ck17sl83c9gya0b17dcvttzm4',
          name: 'Dummy',
          role: 'BUDDY',
          photo: null,
          newbies: [
            {
              id: 'ck17svulh9k2k0b17j31ansfk',
              photo: null,
              name: 'Test 2',
              position: 'Front End',
              startDate: null,
              newbieTasks: [
                {
                  id: '1',
                  status: TaskStatus.Completed,
                },
              ],
              buddyTasks: [
                {
                  id: '1',
                  status: TaskStatus.Uncompleted,
                },
              ],
            },
            {
              id: 'ck17swp2m9kcv0b17we0ibrdn',
              photo: null,
              name: 'Test 1',
              position: 'Front End',
              startDate: null,
              newbieTasks: [
                {
                  id: '1',
                  status: TaskStatus.Completed,
                },
              ],
              buddyTasks: [
                {
                  id: '1',
                  status: TaskStatus.Uncompleted,
                },
              ],
            },
          ],
        },
      },
    },
  },
];

export const buddySelectMock = [
  {
    request: {
      query: BUDDY_SELECT,
      variables: {
        talentId: 'ck17sl83c9gya0b17dcvttzm4',
      },
    },
    result: {
      data: {
        talent: {
          id: 'ck17sl83c9gya0b17dcvttzm4',
          name: 'Dummy',
          role: 'BUDDY',
          photo: null,
          buddies: [],
        },
      },
    },
  },
];

export const taskDetailsMock = [
  {
    request: {
      query: TASK_DETAILS,
      variables: {
        taskId: '1',
      },
    },
    result: {
      data: {
        task: {
          id: '1',
          title: 'New task',
          description: 'New task',
          status: TaskStatus.Uncompleted,
          newbie: getBasicUserDetailsMock(),
        },
      },
    },
  },
];

export const addTaskSuccessMock = (
  variables?: Partial<MutationAddNewbieTaskArgs>,
  data?: Partial<AuthPayload>
) => ({
  request: {
    query: ADD_NEWBIE_TASK,
    variables: {
      newbieId: '1234',
      input: {
        title: 'Test task',
        description: '<h1>Hello world!</h1>',
      },
      ...variables,
    },
  },
  result: {
    data: {
      addNewbieTask: {
        id: 'ck3sw1h1mqkd90964u4675qo2',
        title: 'Test task',
        newbie: {
          id: '1',
          newbieTasks: [],
        },
        ...data,
      },
    },
  },
});

export const addTaskFailedMock = (
  variables?: Partial<MutationAddNewbieTaskArgs>
) => ({
  request: {
    query: ADD_NEWBIE_TASK,
    variables: {
      newbieId: '1234',
      input: {
        title: 'Test task',
        description: '<h1>Hello world!</h1>',
      },
      ...variables,
    },
  },
  result: {
    errors: [new GraphQLError('No such user')],
  },
});

export const addNewbieSuccessMock = () => ({
  request: {
    query: ADD_NEWBIE,
    variables: {
      buddyId: '1234',
      input: {
        email: 'test_newbie@email.com',
        name: 'test newbie',
      },
    },
  },
  result: {
    data: {
      addNewbie: {
        id: 'ck3sw1h1mqkd90964u4675qo2',
        email: 'test_newbie@email.com',
        name: 'test newbie',
        role: UserRole.Newbie,
      },
    },
  },
});

export const addNewbieFailedMock = () => ({
  request: {
    query: ADD_NEWBIE,
    variables: {
      buddyId: '1234',
      input: {
        email: 'test_newbie@email.com',
        name: 'test newbie',
      },
    },
  },
  result: {
    errors: [new GraphQLError('No such user')],
  },
});
