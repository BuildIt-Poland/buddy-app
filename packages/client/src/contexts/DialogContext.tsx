import React, { useState, useContext } from 'react';
import AlertDialog from 'atoms/AlertDialog';

interface State {
  isOpen: boolean;
  message: string;
  title: string | undefined;
  onConfirm?: () => void;
}

interface DialogContextData {
  showDialog: (message: string, title?: string, onConfirm?: () => void) => void;
  hideDialog: () => void;
}

const defaultState: State = {
  isOpen: false,
  title: '',
  message: '',
};

const DialogContext = React.createContext<DialogContextData | undefined>(undefined);

interface DialogProviderProps {
  children: React.ReactNode;
  value?: DialogContextData;
}

const DialogProvider = (props: DialogProviderProps): JSX.Element => {
  const [state, setDialog] = useState<State>(defaultState);

  const showDialog = (message: string, title?: string, onConfirm?: () => void) =>
    setDialog({ isOpen: true, message, title, onConfirm });

  const hideDialog = () => setDialog(defaultState);

  const onConfirmHandler = () => {
    state.onConfirm && state.onConfirm();
    hideDialog();
  };

  return (
    <DialogContext.Provider
      value={{
        showDialog,
        hideDialog,
      }}>
      <AlertDialog
        isOpen={state.isOpen}
        message={state.message}
        onConfirm={state.onConfirm ? onConfirmHandler : undefined}
        title={state.title}
        onClose={hideDialog}
      />
      {props.children}
    </DialogContext.Provider>
  );
};

const useDialog = () => {
  const context = useContext(DialogContext);
  if (context === undefined) {
    throw new Error(`useDialog must be used within a DialogProvider`);
  }
  return context;
};

export { DialogProvider, useDialog };
