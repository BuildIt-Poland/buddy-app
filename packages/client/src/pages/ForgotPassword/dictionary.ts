interface ForgotPasswordDictionary {
  TITLE: string;
  TO_LOGIN: string;
  SUBMIT: string;
  EMAIL: {
    LABEL: string;
    REQUIRED: string;
    INVALID: string;
  };
  DIALOG: {
    SUCCESS_TITLE: string;
    SUCCESS_MSG: string;
  };
  ERRORS: {
    NO_NETWORK: string;
  };
}

const DICTIONARY: ForgotPasswordDictionary = {
  TITLE: 'Buddy',
  TO_LOGIN: 'Back to login',
  SUBMIT: 'Submit',
  EMAIL: {
    LABEL: 'Email Address',
    REQUIRED: 'Email address is required',
    INVALID: 'Please enter a valid email',
  },
  DIALOG: {
    SUCCESS_TITLE: 'Confirmation',
    SUCCESS_MSG:
      'The email has been sent successfully. \nPlease check your account for a password reset link',
  },
  ERRORS: {
    NO_NETWORK:
      'Could not connect to the server. \n Please verify your internet connection.',
  },
};

export default DICTIONARY;
