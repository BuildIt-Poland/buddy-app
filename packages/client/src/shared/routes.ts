enum ROUTES {
  BASE = '/',
  LOGIN = '/login',
  FORGOT_PASSWORD = '/forgot-password',
  RESET_PASSWORD = '/reset-password/:token',
  ROUTE_404 = '/404',

  TALENT = '/talent',
  TALENT_SELECT_BUDDY = '/talent/buddies',
  TALENT_SELECT_NEWBIE = '/talent/buddies/:buddyId/newbies',
  TALENT_ADD_TASK = '/talent/newbies/:newbieId/add-task',
  TALENT_TASK_DETAILS = '/talent/newbies/:newbieId/tasks/:taskId',
  TALENT_TASKS_LIST = '/talent/newbies/:newbieId/tasks',
  TALENT_ADD_NEWBIE = '/talent/buddies/:buddyId/add-newbie',
  TALENT_ADD_BUDDY = '/talent/add-buddy',
  TALENT_ADD_TALENT = '/talent/add-talent',
  TALENT_DETAILS = '/talent/details',
  TALENT_BUDDY_DETAILS = '/talent/buddies/:buddyId/details',
  TALENT_NEWBIE_DETAILS = '/talent/newbies/:newbieId/details',

  BUDDY = '/buddy/',
  BUDDY_DETAILS = '/buddy/details',
  BUDDY_ADD_TASK = '/buddy/newbies/:newbieId/add-task',
  BUDDY_TASK_DETAILS = '/buddy/newbies/:newbieId/tasks/:taskId',
  BUDDY_TASKS_LIST = '/buddy/newbies/:newbieId/tasks',
  BUDDY_SELECT_NEWBIE = '/buddy/newbies',
  BUDDY_ADD_NEWBIE = '/buddy/add-newbie',
  BUDDY_NEWBIE_DETAILS = '/buddy/newbies/:newbieId/details',

  NEWBIE = '/newbie',
  NEWBIE_DETAILS = '/newbie/details',
  NEWBIE_TASKS_LIST = '/newbie/tasks',
  NEWBIE_TASK_DETAILS = '/newbie/tasks/:taskId',
  NEWBIE_BUDDY_DETAILS = '/newbie/buddy/:buddyId/details',
}

export { ROUTES };
