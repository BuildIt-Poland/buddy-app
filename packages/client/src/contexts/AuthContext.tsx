import React, { useReducer, useEffect, useContext } from 'react';
import { Mutation, AuthPayload } from '@buddy-app/schema';
import { useQuery } from '@apollo/react-hooks';
import { ExecutionResult } from '@apollo/react-common';
import { auth } from 'utils';
import { LOGIN_MUTATION } from 'graphql/login.graphql';
import { GET_LOCAL_AUTH } from 'graphql/get-auth.graphql';
import { AuthCache, setCacheToken } from 'utils/apollo-client/cache';
import { apolloClient } from 'utils';

enum ActionTypes {
  AUTH_INIT = 'auth/init',
  AUTH_ERROR = 'auth/error',
  AUTH_SUCCESS = 'auth/success',
  AUTH_LOGOUT = 'auth/logout',
}

interface Action {
  type: ActionTypes;
  payload?: AuthPayload;
}

export interface AuthState {
  loading: boolean;
  error?: any;
  data: AuthPayload;
  isAuthenticated: boolean;
}

const defaultState: AuthState = {
  loading: false,
  error: null,
  data: {} as AuthPayload,
  isAuthenticated: false,
};

interface AuthContext extends AuthState {
  login: (email: string, password: string) => Promise<ExecutionResult<Mutation>>;
  logout: () => void;
}

interface AuthProviderProps {
  children: React.ReactNode;
  value?: AuthContext;
}

const defaultContext: AuthContext = {
  ...defaultState,
  login: () => Promise.reject(),
  logout: () => null,
};

const AuthContext = React.createContext(defaultContext);

const authReducer = (state: AuthState, action: Action): AuthState => {
  switch (action.type) {
    case ActionTypes.AUTH_INIT:
      return {
        ...state,
        loading: true,
        error: null,
      };
    case ActionTypes.AUTH_SUCCESS:
      return {
        ...state,
        loading: false,
        data: { ...action.payload } as AuthPayload,
        isAuthenticated: true,
      };
    case ActionTypes.AUTH_ERROR:
      return {
        ...state,
        loading: false,
        error: { ...action.payload },
      };
    case ActionTypes.AUTH_LOGOUT:
      return {
        ...state,
        data: {} as AuthPayload,
        isAuthenticated: false,
      };
    default:
      throw new Error('Not valid action');
  }
};

const AuthProvider = ({ children, ...props }: AuthProviderProps) => {
  const [state, dispatch] = useReducer(authReducer, defaultState);
  const { data } = useQuery<AuthCache>(GET_LOCAL_AUTH);

  useEffect(() => {
    const user = auth.getUser();
    if (user) {
      dispatch({
        type: ActionTypes.AUTH_SUCCESS,
        payload: user,
      });
    }
  }, []);

  useEffect(() => {
    if (data && data.auth && data.auth.tokenHasExpired) {
      logout();
    }
  }, [data]);

  const login = async (email: string, password: string) => {
    dispatch({ type: ActionTypes.AUTH_INIT });

    try {
      const { data } = await apolloClient.mutate<Mutation>({
        mutation: LOGIN_MUTATION,
        variables: {
          email,
          password,
        },
      });

      if (data) {
        auth.setUser(data.login);
        dispatch({
          type: ActionTypes.AUTH_SUCCESS,
          payload: data.login,
        });
      }

      return Promise.resolve({ data });
    } catch (error) {
      dispatch({ type: ActionTypes.AUTH_ERROR, payload: error });
      return Promise.reject(error);
    }
  };

  const logout = () => {
    auth.removeUser();
    apolloClient.writeData({
      data: {
        ...setCacheToken(false),
      },
    });
    dispatch({ type: ActionTypes.AUTH_LOGOUT });
  };

  return (
    <AuthContext.Provider value={{ ...state, login, logout }} {...props}>
      {children}
    </AuthContext.Provider>
  );
};

const useAuth = () => {
  const context = useContext(AuthContext);
  if (context === undefined) {
    throw new Error(`useAuth must be used within a AuthProvider`);
  }
  return context;
};

export { AuthProvider, useAuth };
