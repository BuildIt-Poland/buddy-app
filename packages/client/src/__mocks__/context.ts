import { AuthState } from 'contexts/AuthContext';
import { UserRole } from '.';

export const mockedTalentContext = (context?: Partial<AuthState>) => ({
  isAuthenticated: true,
  loading: false,
  data: {
    role: UserRole.Talent,
    token: 'token',
    userId: '1234',
  },
  ...context,
});

export const mockedBuddyContext = (context?: Partial<AuthState>) => ({
  isAuthenticated: true,
  loading: false,
  data: {
    role: UserRole.Buddy,
    token: 'token',
    userId: '1234',
  },
  ...context,
});

export const mockedNewbieContext = (context?: Partial<AuthState>) => ({
  isAuthenticated: true,
  loading: false,
  data: {
    role: UserRole.Newbie,
    token: 'token',
    userId: '1234',
  },
  ...context,
});
