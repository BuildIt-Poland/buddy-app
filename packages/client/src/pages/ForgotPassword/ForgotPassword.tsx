import React from 'react';
import { Link, useHistory } from 'react-router-dom';
import { useMutation } from '@apollo/react-hooks';
import { Mutation } from '@buddy-app/schema';
import { FORGOT_PASSWORD_MUTATION } from 'graphql/forgot-password.graphql';
import TextField from '@material-ui/core/TextField';
import Grid from '@material-ui/core/Grid';
import { makeStyles } from '@material-ui/core/styles';
import CircularProgress from '@material-ui/core/CircularProgress';
import { useForm } from 'react-hook-form';
import AuthContainer from 'atoms/AuthContainer';
import { useDialog } from 'contexts/DialogContext';
import RoundedButton from 'atoms/RoundedButton';
import { auth } from 'utils';
import { ROUTES } from 'shared/routes';
import DICTIONARY from './dictionary';
import { FormData } from './types';

const useStyles = makeStyles(theme => ({
  form: {
    width: '100%',
  },
  submit: {
    margin: theme.spacing(3, 0),
  },
  toLogin: {
    fontSize: '1.5rem',
    '&:hover': {
      color: theme.palette.action.active,
    },
  },
}));

const ForgotPassword: React.FC = () => {
  const classes = useStyles();
  const history = useHistory();
  const { register, errors, handleSubmit } = useForm<FormData>();
  const { showDialog } = useDialog();
  const url = `${window.location.origin}/reset-password`;

  const onCompleted = (data?: any) => {
    if (data) {
      auth.setForgotPasswordUser(data.sendResetPasswordLink);
    }
    showDialog(DICTIONARY.DIALOG.SUCCESS_MSG, DICTIONARY.DIALOG.SUCCESS_TITLE);
    history.push(ROUTES.LOGIN);
  };

  const [sendResetPasswordLink, { loading }] = useMutation<Partial<Mutation>>(
    FORGOT_PASSWORD_MUTATION,
    {
      onCompleted,
      onError: error => {
        if (error) {
          if (error.networkError) {
            showDialog(DICTIONARY.ERRORS.NO_NETWORK);
          } else {
            onCompleted();
          }
        }
      },
    }
  );

  const onFormChange = (e: React.ChangeEvent<HTMLFormElement>) =>
    (e.target.value = e.target.value.trim());

  const onSubmit = ({ email }: FormData) => {
    sendResetPasswordLink({
      variables: {
        email,
        url,
      },
    });
  };

  return (
    <AuthContainer data-testid='forgot-password-page' title={DICTIONARY.TITLE}>
      <form
        className={classes.form}
        data-testid='form'
        noValidate
        onChange={onFormChange}
        onSubmit={handleSubmit(onSubmit)}>
        <TextField
          inputProps={{ 'data-testid': 'email' }}
          inputRef={register({
            required: DICTIONARY.EMAIL.REQUIRED,
            pattern: {
              value: /^[A-Z0-9._%+-]+@[A-Z0-9.-]+\.[A-Z]{2,4}$/i,
              message: DICTIONARY.EMAIL.INVALID,
            },
          })}
          margin={'dense'}
          fullWidth
          label={DICTIONARY.EMAIL.LABEL}
          name='email'
          autoComplete='email'
          autoFocus
          error={!!errors.email}
          helperText={(errors.email && errors.email.message) || ' '}
        />
        <RoundedButton
          data-testid='submit-button'
          type='submit'
          fullWidth
          variant='contained'
          color='primary'
          disabled={loading}
          className={classes.submit}>
          {loading ? (
            <CircularProgress
              data-testid='login-progress'
              variant={'indeterminate'}
              size={32}
            />
          ) : (
            DICTIONARY.SUBMIT
          )}
        </RoundedButton>
        <Grid container justify='flex-end'>
          <Link to={ROUTES.LOGIN} className={classes.toLogin}>
            {DICTIONARY.TO_LOGIN}
          </Link>
        </Grid>
      </form>
    </AuthContainer>
  );
};

export default ForgotPassword;
