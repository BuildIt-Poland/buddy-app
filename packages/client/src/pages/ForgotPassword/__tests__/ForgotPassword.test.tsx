import React from 'react';
import { create, act } from 'react-test-renderer';
import waitForExpect from 'wait-for-expect';
import { MemoryRouter, Route } from 'react-router-dom';
import { MockedProvider } from '@apollo/react-testing';
import { mockLocation } from '__mocks__';

import { ROUTES } from 'shared/routes';
import { AuthProvider } from 'contexts/AuthContext';
import { DialogProvider } from 'contexts/DialogContext';
import ForgotPassword from '../ForgotPassword';

jest.mock('utils/auth');
jest.mock('utils/apollo-client');

describe('Component - ForgotPassword', () => {
  const component = create(
    <MockedProvider mocks={[]} addTypename={false} resolvers={{}}>
      <MemoryRouter initialEntries={[mockLocation(ROUTES.FORGOT_PASSWORD)]}>
        <AuthProvider>
          <DialogProvider>
            <Route path={ROUTES.FORGOT_PASSWORD} component={ForgotPassword} />
          </DialogProvider>
        </AuthProvider>
      </MemoryRouter>
    </MockedProvider>
  );

  it('renders correctly', async () => {
    await act(async () => {
      await waitForExpect(() => {
        expect(component.toJSON()).toMatchSnapshot();
      });
    });
  });
});
