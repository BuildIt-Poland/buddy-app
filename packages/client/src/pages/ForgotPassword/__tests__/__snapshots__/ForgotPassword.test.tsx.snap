// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component - ForgotPassword renders correctly 1`] = `
<main
  className="MuiContainer-root makeStyles-container-22 makeStyles-container-20 MuiContainer-maxWidthMd"
>
  <h1
    className="MuiTypography-root makeStyles-title-21 MuiTypography-h1 MuiTypography-alignCenter"
  >
    Buddy
  </h1>
  <svg
    className="makeStyles-spaceMan-19"
  >
    spaceman.svg
  </svg>
  <form
    className="makeStyles-form-16"
    data-testid="form"
    noValidate={true}
    onChange={[Function]}
    onSubmit={[Function]}
  >
    <div
      className="MuiFormControl-root MuiTextField-root MuiFormControl-marginDense MuiFormControl-fullWidth"
    >
      <label
        className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-marginDense"
        data-shrink={false}
      >
        Email Address
      </label>
      <div
        className="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-fullWidth MuiInput-fullWidth MuiInputBase-formControl MuiInput-formControl MuiInputBase-marginDense MuiInput-marginDense"
        onClick={[Function]}
      >
        <input
          aria-invalid={false}
          autoComplete="email"
          autoFocus={true}
          className="MuiInputBase-input MuiInput-input MuiInputBase-inputMarginDense MuiInput-inputMarginDense"
          data-testid="email"
          disabled={false}
          name="email"
          onAnimationStart={[Function]}
          onBlur={[Function]}
          onChange={[Function]}
          onFocus={[Function]}
          required={false}
          type="text"
        />
      </div>
      <p
        className="MuiFormHelperText-root MuiFormHelperText-marginDense"
      >
        <span
          dangerouslySetInnerHTML={
            Object {
              "__html": "&#8203;",
            }
          }
        />
      </p>
    </div>
    <button
      className="MuiButtonBase-root MuiButton-root WithStyles(ForwardRef(Button))-root-126 MuiButton-contained makeStyles-submit-17 MuiButton-containedPrimary MuiButton-fullWidth"
      data-testid="submit-button"
      disabled={false}
      onBlur={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex={0}
      type="submit"
    >
      <span
        className="MuiButton-label"
      >
        Submit
      </span>
      <span
        className="MuiTouchRipple-root"
      />
    </button>
    <div
      className="MuiGrid-root MuiGrid-container MuiGrid-justify-xs-flex-end"
    >
      <a
        className="makeStyles-toLogin-18"
        href="/login"
        onClick={[Function]}
      >
        Back to login
      </a>
    </div>
  </form>
  <svg
    className="makeStyles-background-262"
  >
    <ellipse
      cx="70%"
      cy="300"
      rx="75%"
      ry="227"
    />
  </svg>
</main>
`;
